### PR TITLE
New Bubblegum loot: Rabid Cuffs

### PR DESCRIPTION
**Pull request by LazennG**
# Document the changes in your pull request

New bubblegum drop: rabid cuffs, gloves that allow the user to target a tile to damage creatures for 30 brute and gib dead ones to recover health. People hit by the ability take 10 damage, and hitting dead ones will restore much less to the user unless they have over a total of 300 brute+burn, where they'll heal for half of eating a simplemob. The move has a 10 second cooldown and has targeting similar to eye of god.

# Wiki Documentation

Under Bubblegum's loot section in the megafauna page:
cuffs

Rabid Cuffs: Gloves that allow the wearer to heal by consuming dead creatures.

# Changelog
:cl:  LazennG @Chubbygummibear @cuackles @UselessTheremin @ToasterBiome @KillerOrcaCora 
rscadd: Added rabid cuffs   
imageadd: added cuff sprites 
/:cl:
thanks to @Chubbygummibear for code help and to @cuackles and @UselessTheremin for helping me do sprites 
also thanks @ToasterBiome for helping me how to sprite at a level above braindead infant and @KillerOrcaCora for making the cuffs visible

### CODE REVIEW

**Comment by Mqiib** [comment]
So it's a free gibber if you space someone

**Comment by LazennG** [comment]
> So it's a free gibber if you space someone

about as free as drake form or the goat plush and a bit less free than the soulstone shard yeah

**Comment by ToasterBiome** [comment]
tfw no credit for advisory role

**Comment by Marmio64** [comment]
the ability to gib people sounds really dumb. Like, you're not going to use it unless you're specifically doing really antaggy or validhunty shit like hunting down changelings just for the hell of it so you can click them for 30 seconds to get them to 500 brute before gibbing them and teabagging the remains. Really should only be able to gib animals and shit.

**Comment by LazennG** [comment]
> the ability to gib people sounds really dumb. Like, you're not going to use it unless you're specifically doing really antaggy or validhunty shit like hunting down changelings just for the hell of it so you can click them for 30 seconds to get them to 500 brute before gibbing them and teabagging the remains. Really should only be able to gib animals and shit.

legion bodies in lavaland meet the gibbing criteria and the ability has a 15 second cooldown while doing 10 damage per use to people

**Comment by ToasterBiome** [comment]
> > the ability to gib people sounds really dumb. Like, you're not going to use it unless you're specifically doing really antaggy or validhunty shit like hunting down changelings just for the hell of it so you can click them for 30 seconds to get them to 500 brute before gibbing them and teabagging the remains. Really should only be able to gib animals and shit.
> 
> legion bodies in lavaland meet the gibbing criteria and the ability has a 15 second cooldown while doing 10 damage per use to people

Maybe as a compromise only let it gib human bodies that are SSD?

**Comment by MusselMain** [comment]
> > > the ability to gib people sounds really dumb. Like, you're not going to use it unless you're specifically doing really antaggy or validhunty shit like hunting down changelings just for the hell of it so you can click them for 30 seconds to get them to 500 brute before gibbing them and teabagging the remains. Really should only be able to gib animals and shit.
> > 
> > 
> > legion bodies in lavaland meet the gibbing criteria and the ability has a 15 second cooldown while doing 10 damage per use to people
> 
> Maybe as a compromise only let it gib human bodies that are SSD?

Better yet make the process of gibbing make very obvious noises and take like 2-3 minutes, this way it's a rewarding, yet risky feature

**Comment by MusselMain** [comment]
> So it's a free gibber if you space someone
"free" as in you have to win a fight versus a Lavaland boss that use to consistently wipe the entire station plus ERT deathsquad without going below a quarter health. I could be wrong but I feel like if you can win that fight, you honestly deserve half decent gear.

**Comment by MusselMain** [comment]
> > So it's a free gibber if you space someone

>So it's a free gibber if you space someone
"free" as in you have to win a fight versus a Lavaland boss that use to consistently wipe the entire station plus ERT deathsquad while it only loses a quarter of it's health. I could be wrong but I feel like if you can win that fight, you honestly deserve half decent gear.

**Comment by Marmio64** [comment]
> > the ability to gib people sounds really dumb. Like, you're not going to use it unless you're specifically doing really antaggy or validhunty shit like hunting down changelings just for the hell of it so you can click them for 30 seconds to get them to 500 brute before gibbing them and teabagging the remains. Really should only be able to gib animals and shit.
> 
> legion bodies in lavaland meet the gibbing criteria and the ability has a 15 second cooldown while doing 10 damage per use to people

If all you want is the ability to gib legion corpses then add a soulless check. There is zero non validhunt uses for being able to gib people who still have souls.

**Comment by LazennG** [comment]
> > > the ability to gib people sounds really dumb. Like, you're not going to use it unless you're specifically doing really antaggy or validhunty shit like hunting down changelings just for the hell of it so you can click them for 30 seconds to get them to 500 brute before gibbing them and teabagging the remains. Really should only be able to gib animals and shit.
> > 
> > 
> > legion bodies in lavaland meet the gibbing criteria and the ability has a 15 second cooldown while doing 10 damage per use to people
> 
> If all you want is the ability to gib legion corpses then add a soulless check. There is zero non validhunt uses for being able to gib people who still have souls.

think about how often you see people with 500 brute damage

**Comment by tattax** [comment]
> Better yet make the process of gibbing make very obvious noises and take like 2-3 minutes, this way it's a rewarding, yet risky feature

my brother in christ

**Comment by JamieD1** [comment]
@SomeguyManperson

**Comment by Marmio64** [comment]
> > > > the ability to gib people sounds really dumb. Like, you're not going to use it unless you're specifically doing really antaggy or validhunty shit like hunting down changelings just for the hell of it so you can click them for 30 seconds to get them to 500 brute before gibbing them and teabagging the remains. Really should only be able to gib animals and shit.
> > > 
> > > 
> > > legion bodies in lavaland meet the gibbing criteria and the ability has a 15 second cooldown while doing 10 damage per use to people
> > 
> > 
> > If all you want is the ability to gib legion corpses then add a soulless check. There is zero non validhunt uses for being able to gib people who still have souls.
> 
> think about how often you see people with 500 brute damage

Not often, which is why it'd specifically be used to get rid of antags by validhunters, since the person needs to go out of their way to really fuck the body up so they can gib it. Like I said, there's just no reason for miners to need this to be able to gib bodies with souls in them unless you are specifically catering this to being a validhunt/antag weapon, which I don't know why you would.

**Comment by LazennG** [comment]
> Not often, which is why it'd specifically be used to get rid of antags by validhunters, since the person needs to go out of their way to really fuck the body up so they can gib it. Like I said, there's just no reason for miners to need this to be able to gib bodies with souls in them unless you are specifically catering this to being a validhunt/antag weapon, which I don't know why you would.

yes of course because they'd go for getting someone to 500 brute damage and gibbing them instead of shuttle crushing or throwing at a megafauna/chasm/gibtonite or any of the many easily accessible gibbing methods that exist in the game. the reason i included gibbing for people like this was for flavor and very limited utility. i disagree with your idea to babyproof the item and think that it's in line with your level of ideas like "instead of making eye of god help with killing fauna, just have it tell the user what's inside the tendril/fauna and nothing else" and (excuse me if i'm mixing you up with someone else) "make the eye of god damage boost not affect people." if validhunting is the only thing you can argue with then i don't think you can compel me to change anything.

**Comment by LazennG** [comment]
After hearing from people who can do more than repeat the same flawed logic over and over again I have decided to remove the gibbing aspect of the item and instead make it heal the user a bit more if the victim is bruised/burnt the fuck up

**Comment by Marmio64** [comment]
the passive aggressiveness is appreciated

**Comment by ToasterBiome** [comment]
> the passive aggressiveness is appreciated

lol

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 1]
```diff
@@ -92,3 +92,96 @@
 		to_chat(M, span_warning("You don't want to hurt anyone, just give them hugs!"))
 		M.a_intent = INTENT_HELP
 	.= FALSE
+
+/obj/item/clothing/gloves/bracer/cuffs
+	name = "rabid cuffs"
+	desc = "Chainless manacles fashioned after the hungriest of slaughter demons. Grants the wearer a similar hunger that can be sated in a similar way."
```
Comment (Anvilman6) [inline code/modules/clothing/gloves/miscellaneous.dm]:
```suggestion
	desc = "Chainless manacles, fashioned in the form of the hungriest of slaughter demons. Wearing these invokes a hunger in the wearer that can only be sated by bloodshed."
```

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 2]
```diff
@@ -92,3 +92,96 @@
...
+	if(ishuman(user) && slot == ITEM_SLOT_GLOVES)
+		user.AddAbility(swipe_ability)
+
+/obj/item/clothing/gloves/bracer/cuffs/dropped(mob/living/user)
+	. = ..()
+	user.RemoveAbility(swipe_ability)
+
+obj/effect/proc_holder/swipe
+	name = "swipe"
+	desc = "Swipe at a target area, dealing damage and consuming dead creatures to heal yourself. People are ineligible for total consumption. Creatures take 30 damage and heal the most while people and cyborgs take 10 damage and heal for the least. People that have been thoroughly burned and bruised heal you for a bit more than those that aren't."
```
Comment (Anvilman6) [inline code/modules/clothing/gloves/miscellaneous.dm]:
```suggestion
	desc = "Swipe at a target area, dealing damage and consuming dead entities to heal yourself. People are ineligible for total consumption. Creatures take 30 damage, and give the most healing, while people and cyborgs take 10 damage and heal for the least. People who have been thoroughly burned and bruised heal you for slightly more!"
```

Comment (Anvilman6) [reply code/modules/clothing/gloves/miscellaneous.dm]: Is the damage intended to be applied to corpses? Or is this on living targets? It's not entirely clear, as you talk about swiping and then consuming dead things in the start. May want to reorder the tooltip into a section about damage, and interactions with living entities, then corpse interactions and healing.

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 3]
```diff
@@ -92,3 +92,96 @@
...
+	if(!isliving(usr))
+		return TRUE
+	var/mob/living/user = usr
+	fire(user)
+
+/obj/effect/proc_holder/swipe/fire(mob/living/carbon/user)
+	if(active)
+		remove_ranged_ability(span_notice("You relax your arms."))
+	else
+		add_ranged_ability(user, span_notice("You ready your cuffs. <B>Left-click a creature or floor to swipe at it!</B>"), TRUE)
```
Comment (Anvilman6) [inline code/modules/clothing/gloves/miscellaneous.dm]:
```suggestion
		add_ranged_ability(user, span_notice("You ready your cuffs. <B>Left-click a creature or nearby location to swipe at it!</B>"), TRUE)
```

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 4]
```diff
@@ -92,3 +92,96 @@
...
+			L.adjustBruteLoss(30)
+			if(L.stat == DEAD)
+				L.gib()
+				caller.adjustBruteLoss(-20)
+				caller.adjustFireLoss(-20)
+				caller.adjustToxLoss(-20)
+				caller.blood_volume = BLOOD_VOLUME_NORMAL(caller)*1.10
+		L.adjustBruteLoss(10)
+		if(L.getBruteLoss()+L.getFireLoss() >= 299)
+			to_chat(caller, span_notice("You're able to consume a bit more as the body has been softened up!"))
```
Comment (Anvilman6) [inline code/modules/clothing/gloves/miscellaneous.dm]:
```suggestion
			to_chat(caller, span_notice("You're able to consume a bit more of the body, as it was previously softened up!"))
```

**Review by Anvilman6 — CHANGES_REQUESTED** [review]
Review moment

File: code/modules/clothing/gloves/miscellaneous.dm (hunk 2 above)
Comment (LazennG) [reply code/modules/clothing/gloves/miscellaneous.dm]: the damage is intended for whoever the attack hits regardless of living or dead status, the conuming thing is just for dead targets; will get to work on making the description clearer though

**Comment by KillerOrcaCora** [comment]
Can you get a screenshot with the new sprites equipped, thanks.

**Comment by LazennG** [comment]
> Can you get a screenshot with the new sprites equipped, thanks.

image
image
image
image

**Review by KillerOrcaCora — APPROVED** [review]
Nice sprite.

**Pull request merged**